tests/Functional/CorporateFooterRenderTest.php

```diff
@@ -0,0 +1,419 @@
+    $this->container->get('theme_handler')->setDefault('oe_theme');
+    $this->container->set('theme.registry', NULL);
+
+    // Rebuild the ui_pattern definitions to collect the ones provided by
+    // oe_theme itself.
+    $this->container->get('plugin.manager.ui_patterns')->clearCachedDefinitions();
+    $this->configFactory = $this->container->get('config.factory');
+  }
+
+  /**
+   * Test corporate footer block rendering.
+   *
+   * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+   * @SuppressWarnings(PHPMD.NPathComplexity)
+   */
+  public function testCorporateFooterRendering(): void {
+    // First test European Commission footer core block rendering.
+    $data = $this->getFixtureContent('ec_footer.yml');
+    $this->overrideCorporateBlocksFooter('ec', $data);
+
+    $this->drupalGet('<front>');
+    $assert = $this->assertSession();
+
+    // Make sure that footer block is present.
+    $this->assertFooterPresence('core', 4);
+
+    $section = $assert->elementExists('css', 'footer.ecl-footer-core section.ecl-footer-core__section1');
+
+    $actual = $section->find('css', 'a.ecl-footer-core__title');
+    $this->assertEquals($data['corporate_site_link']['label'], $actual->getText());
+    $this->assertEquals($data['corporate_site_link']['href'], $actual->getAttribute('href'));
+
+    // Site owner is not set yet, lets make sure we don't have a description.
+    $assert->elementNotExists('css', 'div.ecl-footer-core__description');
+
+    $section = $assert->elementExists('css', 'footer.ecl-footer-core section.ecl-footer-core__section2');
+    $items = $data['class_navigation'];
+
+    foreach ($items as $key => $expected) {
+      $index = $key + 1;
+      $actual = $section->find('css', "ul li:nth-child({$index}) > a");
+      $this->assertListLink($actual, 'core', $expected);
+    }
+
+    $section = $assert->elementExists('css', 'footer.ecl-footer-core section.ecl-footer-core__section3');
+    $items = $data['service_navigation'];
+
+    foreach ($items as $key => $expected) {
+      $index = $key + 1;
+      $actual = $section->find('css', "ul li:nth-child({$index}) > a");
+      $this->assertListLink($actual, 'core', $expected);
+    }
+
+    $section = $assert->elementExists('css', 'footer.ecl-footer-core section.ecl-footer-core__section4');
+    $items = $data['legal_navigation'];
+
+    foreach ($items as $key => $expected) {
+      $index = $key + 1;
+      $actual = $section->find('css', "ul li:nth-child({$index}) > a");
+      $this->assertListLink($actual, 'core', $expected);
+    }
+
+    // Update settings, assert footer changed.
+    $this->updateSiteSettings('http://publications.europa.eu/resource/authority/corporate-body/ACP-EU_JA', 'EC Site Name');
+    $this->drupalGet('<front>');
+
+    $actual = $assert->elementExists('css', 'div.ecl-footer-core__description');
+    $expected = new FormattableMarkup('This site is managed by the @name', ['@name' => 'ACP–EU Joint Assembly']);
```

Comment:
Change all these expected values into simple string, you don't need to pass it as a formattable markup class.